### PR TITLE
feature flags

### PR DESCRIPTION
**Pull request by Rylan-Meilutis**
changed feature flags slightly

### CODE REVIEW

File: 2020-InfiniteRecharge/src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -225,38 +228,38 @@ public void disabledPeriodic()
   public void initialize()
   {
     robotInitialized = true;
-    if(ImuEnable)
+    if(ImuEnable && robotInitialized != false)
```
Comment (mmarchetti) [inline 2020-InfiniteRecharge/src/main/java/frc/robot/Robot.java]: The purpose of the `robotInitialized != false` isn't clear - you've set it to `true` just above here.

File: 2020-InfiniteRecharge/src/main/java/frc/robot/Robot.java [hunk 2]
```diff
@@ -186,9 +186,12 @@ public void teleopPeriodic()
       powerCell.feeder();
   }
       //Compressor
-     if(compressorEnable){
+     if(compressorEnable && compressorControl.useCompressor){
       compressorControl.enable();
     }
+    else if(compressorEnable){
+      compressorControl.disable();
+    }
```
Comment (mmarchetti) [inline 2020-InfiniteRecharge/src/main/java/frc/robot/Robot.java]: Can the `compressorControl.useCompressor` check be in an `if/else` inside the `if (compressorEnable)`?

File: 2020-InfiniteRecharge/src/main/java/frc/robot/Robot.java [hunk 3]
```diff
@@ -225,38 +228,38 @@ public void disabledPeriodic()
...
     System.out.println("Initilied");
-    if(chassisEnable)
+    if(chassisEnable && robotInitialized != false)
     {
       driveTrain.Init(oi);
     }
     System.out.println("ChassisEnable: " + chassisEnable);
 
-    if(climberEnable){
+    if(climberEnable && robotInitialized == false){
```
Comment (mmarchetti) [inline 2020-InfiniteRecharge/src/main/java/frc/robot/Robot.java]: Doesn't this check make it so the climber never gets initialized?

File: 2020-InfiniteRecharge/src/main/java/frc/robot/Robot.java (hunk 2 above)
Comment (Rylan-Meilutis) [reply 2020-InfiniteRecharge/src/main/java/frc/robot/Robot.java]: That is now done

File: 2020-InfiniteRecharge/src/main/java/frc/robot/Robot.java (hunk 3 above)
Comment (Rylan-Meilutis) [reply 2020-InfiniteRecharge/src/main/java/frc/robot/Robot.java]: That has been changed as well

File: 2020-InfiniteRecharge/src/main/java/frc/robot/Robot.java (hunk 1 above)
Comment (Rylan-Meilutis) [reply 2020-InfiniteRecharge/src/main/java/frc/robot/Robot.java]: This has been changed

File: 2020-InfiniteRecharge/src/main/java/frc/robot/Robot.java [hunk 4]
```diff
@@ -224,40 +235,46 @@ public void disabledPeriodic()
...
-    if(climberEnable){
+    if(climberEnable && climberEnabled == false){
       climber.ClimberInit(oi);
+      climberEnabled = true;
     }
 
-    if(camera1Enable){
+    if(camera1Enable && camera1Enabled == false){
       camera1.init();
+      camera1Enable = true;
```
Comment (mmarchetti) [inline 2020-InfiniteRecharge/src/main/java/frc/robot/Robot.java:262]: `camera1Enabled = true` ?

File: 2020-InfiniteRecharge/src/main/java/frc/robot/Robot.java [hunk 5]
```diff
@@ -224,40 +235,46 @@ public void disabledPeriodic()
   
   public void initialize()
   {
-    robotInitialized = true;
```
Comment (mmarchetti) [inline 2020-InfiniteRecharge/src/main/java/frc/robot/Robot.java:227]: I see how you've made tracking of the initialization states more fine-grained. This seems like a good idea. Can we remove `robotInitialized` entirely, and call `initialize()` unconditionally in `autonomousInit` and `teleopInit`?

File: 2020-InfiniteRecharge/src/main/java/frc/robot/Robot.java [hunk 6]
```diff
@@ -45,15 +45,27 @@
 
   // feature flags booleans
   private boolean camera1Enable = false;
+  private boolean camera1Enabled = false;
```
Comment (mmarchetti) [inline 2020-InfiniteRecharge/src/main/java/frc/robot/Robot.java]: Small suggestion (a nit really) - these names are pretty similar and could be confusing. Would it be worth renaming to `camera1Initialized` etc?

Comment (Rylan-Meilutis) [reply 2020-InfiniteRecharge/src/main/java/frc/robot/Robot.java]: done

File: 2020-InfiniteRecharge/src/main/java/frc/robot/Robot.java (hunk 5 above)
Comment (Rylan-Meilutis) [reply 2020-InfiniteRecharge/src/main/java/frc/robot/Robot.java:227]: agreed and done

File: 2020-InfiniteRecharge/build.gradle [hunk 7]
```diff
@@ -1,6 +1,6 @@
 plugins {
     id "java"
-    id "edu.wpi.first.GradleRIO" version "2020.1.2"
+    id "edu.wpi.first.GradleRIO" version "2020.2.2"
```
Comment (mmarchetti) [inline 2020-InfiniteRecharge/build.gradle]: We should hold off this upgrade until everyone gets 2020.2.2 WPILib installed.

**Review by mmarchetti — APPROVED** [review]

**Pull request merged**